src/blocks/effects/replaceText.ts

```diff
@@ -33,21 +33,26 @@ export function replaceText({
   replacement: string;
   visited: WeakSet<Node>;
 }) {
-  if (visited.has(node)) {
-    // Avoid running replaceText on same node twice if selector passed to brick matches multiple nodes
-    // in the same subtree
-    return;
-  }
-
-  visited.add(node);
+  // `ownerDocument` is null for documents
+  const nodeDocument = node.ownerDocument ?? (node as Document);
+  const walker = nodeDocument.createTreeWalker(node, NodeFilter.SHOW_TEXT);
 
-  if (node.nodeType === Node.TEXT_NODE) {
-    node.textContent = node.textContent.replaceAll(pattern, replacement);
-    return;
-  }
+  for (
+    let currentNode = walker.nextNode();
+    currentNode;
+    currentNode = walker.nextNode()
+  ) {
```

Comment:
I'd probably go for this instead a `for` loop
```suggestion
	let currentNode;
  // eslint-disable-next-line I'm sure there's a rule preventing this
  while (currentNode = walker.nextNode()) {
```

Other than that, it looks fine.

Reply:
Copilot wrote it, so I'm not particularly attached to it. There is indeed an eslint rule no-cond-assign for that condition. I agree it looks slightly cleaner though

Reply:
Prettier magically eliminated the lint warning -- there's no lint if you wrap the whole expression in parentheses 🤷

Reply:
# BOTS GIVETH
# BOTS TAKETH